### PR TITLE
Bump version number to v0.10

### PR DESCRIPTION
**Pull request by josh146**
Bumps the version number to v0.10.0.

### CODE REVIEW

File: qchem/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Release 0.10.0 (development release)
+# Release 0.10.0 (current release)
```
Comment (josh146) [inline qchem/CHANGELOG.md:1]: @agran2018, can you please check the changelog for QChem, and let me know if it all looks okay 🙂

**Comment by josh146** [comment]
@antalszava @agran2018: I also have a feeling that the pyscf bug that required pinning pyscf to version `<=1.7.1` is now fixed, as of May 20: https://github.com/pyscf/pyscf/issues/618. Was wondering if either of you know if this is the case?

File: pennylane/_version.py [hunk 2]
```diff
@@ -16,4 +16,4 @@
 Version number (major.minor.patch[-label])
 """
 
-__version__ = "0.10.0-dev"
+__version__ = "0.10.0"
```
Comment (mariaschuld) [inline pennylane/_version.py:19]: So this should not be 0.11.0-dev now, since we start merging PRs soon?

Comment (josh146) [reply pennylane/_version.py:19]: As soon as this PR is merged, we can make a GitHub release + tag + upload to pypi. Then, the next commit is to bump to 0.11.0-dev, and the feature freeze is complete 🙂

File: qchem/requirements.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 numpy
 scipy
-pennylane>=0.7.0
+pennylane>=0.10.0
 openfermion==0.10.0
 openfermionpsi4==0.4
-pyscf<=1.7.1
+pyscf
```
Comment (agran2018) [inline qchem/requirements.txt]: I think there were issues with the latest version  of PySCF (undefined references), that's why it was pinned to version 1.7.1

File: qchem/setup.py [hunk 4]
```diff
@@ -16,8 +16,7 @@
 with open("pennylane_qchem/_version.py") as f:
     version = f.readlines()[-1].split()[-1].strip("\"'")
 
-requirements = ["pennylane", "openfermion", "openfermionpyscf",
-        "openfermionpsi4", "pyscf<=1.7.1"]
+requirements = ["pennylane>=0.10", "openfermion", "openfermionpyscf", "openfermionpsi4", "pyscf"]
```
Comment (agran2018) [inline qchem/setup.py]: same comment here related to PySCF

File: qchem/requirements.txt (hunk 3 above)
Comment (josh146) [reply qchem/requirements.txt]: I've reverted it :)

**Pull request merged**